app/services/preserved_object_handler.rb

```diff
@@ -79,28 +79,28 @@ def check_existence
           if incoming_version == pres_copy.version
             handler_results.add_result(PreservedObjectHandlerResults::VERSION_MATCHES, pres_copy.class.name)
             handler_results.add_result(PreservedObjectHandlerResults::VERSION_MATCHES, pres_object.class.name)
-            update_pc_validation_timestamps(pres_copy)
-            update_db_object(pres_copy)
-          else
-            version_comp_result = if incoming_version > pres_copy.version
-                                    PreservedObjectHandlerResults::ARG_VERSION_GREATER_THAN_DB_OBJECT
-                                  else
-                                    PreservedObjectHandlerResults::ARG_VERSION_LESS_THAN_DB_OBJECT
-                                  end
-            handler_results.add_result(version_comp_result, pres_copy.class.name)
-            handler_results.add_result(version_comp_result, pres_object.class.name)
+          elsif incoming_version > pres_copy.version
+            handler_results.add_result(PreservedObjectHandlerResults::ARG_VERSION_GREATER_THAN_DB_OBJECT, pres_copy.class.name)
+            handler_results.add_result(PreservedObjectHandlerResults::ARG_VERSION_GREATER_THAN_DB_OBJECT, pres_object.class.name)
             if moab_validation_errors.empty?
-              update_preserved_copy_version_etc(pres_copy, incoming_version, incoming_size, true)
+              update_preserved_copy_version_etc(pres_copy, incoming_version, incoming_size, false)
               update_status(pres_copy, PreservedCopy::OK_STATUS)
-              update_db_object(pres_copy)
               pres_object.current_version = incoming_version
               update_db_object(pres_object)
             else
               update_status(pres_copy, PreservedCopy::INVALID_MOAB_STATUS)
-              update_pc_validation_timestamps(pres_copy)
-              update_db_object(pres_copy)
+            end
+          else # incoming_version < pres_copy.version
+            handler_results.add_result(PreservedObjectHandlerResults::ARG_VERSION_LESS_THAN_DB_OBJECT, pres_copy.class.name)
+            handler_results.add_result(PreservedObjectHandlerResults::ARG_VERSION_LESS_THAN_DB_OBJECT, pres_object.class.name)
+            if moab_validation_errors.empty?
+              update_status(pres_copy, PreservedCopy::EXPECTED_VERS_NOT_FOUND_ON_STORAGE_STATUS)
+            else
+              update_status(pres_copy, PreservedCopy::INVALID_MOAB_STATUS)
             end
           end
+          update_pc_validation_timestamps(pres_copy)
```

Comment:
ah, ok, this ameliorates the concerns from the two prior comments... but won't this also be hit in the "happy path" (`incoming_version == pres_copy.version` evaluates to true)?  in which case it seems we didn't validate (which i believe is the behavior described by the audit doc)?

Reply:
ah yes - you are correct here.  Hrmm.

Reply:
oh wait - we're fine because in ALL cases, we have in fact done an audit and therefore update audit timestamps.

Reply:
perhaps we are running into more terminology woes.  The timestamps are:

last_audited
last_checked_on_storage

I think maybe we are now wondering about distinctions among:

- last-checksum-audit
- last-moab-validation-audit
- last-version-audit

etc.

oh joy!

Reply:
filed issues #434 to address the follow on work.